common/core/desktop/src/kmx/kmx_context.h

```diff
@@ -23,85 +23,90 @@ class KMX_Context
-   *         have meant splitting a surrogate pair
-   * @param buf      The data buffer to copy current context
-   * @param bufsize  The number of code units ie size of the KMX_WCHAR buffer - not the code points
+   * Copies the characters in CurContext to supplied buffer.
+   * If bufsize is reached before the entire context was copied, the buf
+   * will be truncated to number of valid characters possible with null character
+   * termination. e.g. it will be one code unit less than bufsize if that would
+   * have meant splitting a surrogate pair
+   * @param  buf      The data buffer to copy current context
+   * @param  bufsize  The number of code units ie size of the KMX_WCHAR buffer - not the code points
    */
   void Get(KMX_WCHAR *buf, int bufsize);
 
    /**
-   * @brief Sets the CurContext to the supplied buf character array and updates the pos index.
+   *  Sets the CurContext to the supplied buf character array and updates the pos index.
    *
    * @param buf
    */
   void Set(const KMX_WCHAR *buf);
 
   /**
-   * @brief  Returns a pointer to the character in the current context buffer which
-   *         will have at most n valid xstring units remaining until the null terminating
-   *         character. It will be one code unit less than bufsize if that would
-   *         have meant splitting a surrogate pair or deadkey.
+   * Returns a pointer to the character in the current context buffer which
+   * will have at most n valid xstring units remaining until the null terminating
+   * character. It will be one code unit less than bufsize if that would
+   * have meant splitting a surrogate pair or deadkey.
    *
-   * @param n        The maximum number of valid xstring units (not code points or code units)
-   * @return KMX_WCHAR*  Pointer to the start postion for a buffer of maximum n xstring units
+   * @param  n            The maximum number of valid xstring units (not code points or code units)
+   * @return  KMX_WCHAR*  Pointer to the start postion for a buffer of maximum n xstring units
    */
-  KMX_WCHAR *Buf(int n);
+  KMX_WCHAR *BufMax(int n);
 
   /**
-   * @brief  Returns a pointer to the character in the current context buffer which
-   *         will have at most n code units remaining until the the null terminating character.
-   *         Note: Unlike BufMax there is no checking for truncation of code points.
-   *         e.g. the pointer may point to a code unit that is half of a surrogate pair.
+   * Returns a pointer to the character in the current context buffer which
+   * will have n valid xstring units remaining until the the null terminating character.
+   * OR
+   * Returns NULL if there are less than n valid xstring units in the current context.
+   * Background this was historically for performance during rule evaluation, if there
+   * are not enough characters to compare, don't event attempt the comparison.
+   * Note: Unlike BufMax there is no checking for truncation of code points.
```

Comment:
`Buf` as well as `BufMax` use `decxstr` to move the pointer, so it does check for truncation of code points.

The difference is that `Buf` will return `NULL` if  n xstring units won't fit, whereas `BufMax` will return a pointer for (n-1) xstring units in that case.

At least that's how it's implemented in core, didn't check the Windows implementation.

Reply:
> Buf as well as BufMax use decxstr to move the pointer, so it does check for truncation of code points.

Of the windows `::Buf` uses `decxstr` now as well. There is the old implementation there where it didn't use `decxstr` the line below commented out. So I can update the comment to remove line 87 

>The difference is that Buf will return NULL if n xstring units won't fit, whereas BufMax will return a pointer for (n-1) xstring units in that case.

No the difference is it returns NULL if there are less than n valid xstring units.  Not if n xstring units won't fit. 

As for the line in `::BufMax` 
`f((intptr_t)(q-p) > n) p = incxstr(p); // Copes with deadkey or supplementary pair at start of returned buffer making it too long`

I can't think that this would happen when using `decxstr` to move the pointer.